india_compliance/gst_india/overrides/transaction.py

```diff
@@ -660,17 +663,55 @@ def get_source_state_code(doc):
     return (doc.supplier_gstin or doc.company_gstin)[:2]
 
 
-def validate_backdated_transaction(doc, gst_settings=None, action="create"):
-    if gstr_1_filed_upto := restrict_gstr_1_transaction_for(
+def validate_backdated_transaction(doc, gst_settings=None, action="submit"):
+    gst_settings = gst_settings or frappe.get_cached_doc("GST Settings")
+
+    gstr_1_filed_upto = restrict_gstr_1_transaction_for(
         doc.posting_date, doc.company_gstin, gst_settings
-    ):
+    )
+
+    if gstr_1_filed_upto:
         frappe.throw(
             _(
                 "You are not allowed to {0} {1} as GSTR-1 has been filed upto {2}"
             ).format(action, doc.doctype, frappe.bold(format_date(gstr_1_filed_upto))),
             title=_("Restricted Changes"),
         )
 
+    gstr_1_filed_upto = get_gstr_1_filed_upto(doc.company_gstin)
+    if not gstr_1_filed_upto or getdate(doc.posting_date) > gstr_1_filed_upto:
+        return
+
+    doc_name = get_gst_return_log_name(doc)
+    if not doc_name:
+        return
+
+    gst_return_log = frappe.get_doc("GST Return Log", doc_name)
+
+    url = f"{frappe.utils.get_url()}/app/{doc.doctype.replace(' ', '-').lower()}/{doc.name}"
+    gst_return_log.add_comment(
+        "Comment",
+        f"{doc.doctype} : [{doc.name}]({url}) has been {action} by {frappe.session.user}",
+    )
+
+
+def get_gst_return_log_name(doc):
+    posting_date = getdate(doc.posting_date)
```

Comment:
default last log

Reply:
Objective was to make sense of differences after return is filed. Hence ideally comment should form part of the period where difference will be visible in reconciliation.